### ISSUE TITLE
Assertion failed '!"Incompatible types for gtNewTempAssign"'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#24344]
Found running Windows/arm64 altjit asm diffs:
```
call jit-diff diff --pmi --diff --base --base_root f:\gh\coreclr12 --arch x64 --build Checked --altjit protononjit.dll --frameworks
```

leading to:

```
Assert failure(PID 21744 [0x000054f0], Thread: 20804 [0x5144]): Assertion failed '!"Incompatible types for gtNewTempAssign"' in 'System.Xml.Xsl.XPath.XPathParser`1[Vector`1][System.Numerics.Vector`1[System.Single]]:Parse(ref,ref,int):struct:this' (IL size 107)
    File: f:\gh\coreclr10\src\jit\gentree.cpp Line: 14523
    Image: F:\gh\coreclr10\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe
```

in System.Private.Xml.log

**Comment by BruceForstall** [comment]
@CarolEidt Looks like `Vector<T>` is involved.
cc @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
Note that this also occurs with Windows/x86 PMI asm diffs, with no altjit involved.

cc @tannergooding

**Comment by CarolEidt** [comment]
Thanks @BruceForstall - I've assigned this to myself and will take a look.

**Comment by CarolEidt** [comment]
Fixed with #24377
